scripts/nn/layers/attention.dml

```diff
@@ -0,0 +1,64 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+source("nn/layers/softmax.dml") as softmax
+
+# 1. Create "calculate_score" function, for the beginning without additional parameters -> calculate with dot product.
+
+calculate_scores = function(matrix[double] query, matrix[double] key)
+    return (matrix[double] scores) {
+    /*
+     * dim(query) = AxB
+     * dim(key) = CxB
+     * dim(scores) = AxC
+     */
+    scores = query %*% t(key)
+}
```

Comment:
Here we are unsure how many parameters / choices a user should have. 

E.g: 
- masking
- scaling
- different calculation variants (or is dot enough)

Reply:
Start with the simplest thing you can think of, then we can add arguments later that add the different parts.